packages/chain-adapters/src/utxo/UtxoBaseAdapter.ts

```diff
@@ -301,6 +301,26 @@ export abstract class UtxoBaseAdapter<T extends UtxoChainId> implements IChainAd
 
       const { inputs, outputs } = coinSelectResult
 
+      const account = await this.getAccount(xpub)
+
+      // Always validate next change and receive addresses
+      const nextChangeAddressIndex = account.chainSpecific.nextChangeAddressIndex ?? 0
+      const nextReceiveAddressIndex = account.chainSpecific.nextReceiveAddressIndex ?? 0
+      const nextChangeAddressInput = {
+        address: account.chainSpecific.addresses?.[nextChangeAddressIndex]?.pubkey,
+      }
+      const nextReceiveAddressInput = {
+        address: account.chainSpecific.addresses?.[nextReceiveAddressIndex]?.pubkey,
+      }
+
+      const addresses = [...inputs, ...outputs, nextChangeAddressInput, nextReceiveAddressInput]
+        .map(({ address }) => address)
+        .filter(Boolean) as string[]
+
+      const uniqueAddresses = [...new Set(addresses)]
+
+      await Promise.all(uniqueAddresses.map(validateAddress))
```

Comment:
TRM can take multiple addresses as input so we could refactor the checks to be performed over a response containing multiple addresses, but not for this PR